rules/typescript.js

```diff
@@ -18,7 +18,11 @@ module.exports = {
     '@typescript-eslint/no-unused-vars': 'error',
     '@typescript-eslint/no-useless-constructor': 'error',
     '@typescript-eslint/prefer-for-of': 'error',
-    '@typescript-eslint/prefer-optional-chain': 'error',
+    '@typescript-eslint/prefer-optional-chain': 'off',
```

Comment:
I don't agree with this change. I do think this rule should throw an error

Reply:
for me it's more a matter of preferences. in LP there are some errors because of this rule, some of them can be fixed. But here, for example, I couldn't use optional chain https://github.com/nxt-insurance/landingpage/blob/23b8ac55980f95593ae80bab657a15e077231da8/src/components/pages/RichContentPage.js#L278, it still throws an error. Am I missing something here?

Reply:
you can definitely use optional chain there. Also, the rule was already set to throw an error, which means it is already a standard, therefore I don't think it is ok to just remove it.

Reply:
Well, I've tried and it still throws an error. Maybe it's because I'm calling a function here, not just a variable

<img width="884" alt="Screenshot 2021-08-24 at 12 16 38" src="https://user-images.githubusercontent.com/22943647/130600526-fdbccd9d-d62a-4ee9-ab61-32971bd54230.png">

Reply:
ah okay, I've managed to fix it by changing the type from `Subscription | void` to `Subscription | undefined`

Reply:
the type of subscription is wrong. Void in TS is not the same as in Flow. If a function can return undefined or number for example you should use `() => undefined | number`. Why? Because void is meant in cases where the return value of a function is not meant to be used. A good example is onClick where you do want  `() => void`. 
In this specific case `listenForPage` in `src/cms/dataFetchers.ts` has the return type:  `Subscription<T> | void` and it should be ` Subscription<T> | undefined`

Reply:
> Also, the rule was already set to throw an error, which means it is already a standard, therefore I don't think it is ok to just remove it.

@ilkap07  I didn't realize that the rule is coming from our rule set. I thought it was some of the plugins we extend.
In such case it is indeed a standard (unless the team agrees otherwise; which is still up to a discussion) and therefore should for now be fixed in the LP project.

Reply:
yes, I've turned it on again and fixed errors in the LP